module-ama/src/main/kotlin/org/quiltmc/community/cozy/modules/ama/_Utils.kt

```diff
@@ -0,0 +1,385 @@
+import com.kotlindiscord.kord.extensions.DISCORD_GREEN
+import com.kotlindiscord.kord.extensions.DISCORD_RED
+import com.kotlindiscord.kord.extensions.DISCORD_YELLOW
+import com.kotlindiscord.kord.extensions.builders.ExtensibleBotBuilder
+import com.kotlindiscord.kord.extensions.components.ComponentContainer
+import com.kotlindiscord.kord.extensions.components.components
+import com.kotlindiscord.kord.extensions.components.ephemeralButton
+import com.kotlindiscord.kord.extensions.utils.loadModule
+import dev.kord.common.entity.ButtonStyle
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.behavior.channel.createEmbed
+import dev.kord.core.behavior.channel.createMessage
+import dev.kord.core.behavior.edit
+import dev.kord.core.entity.Message
+import dev.kord.core.entity.User
+import dev.kord.core.entity.channel.GuildMessageChannel
+import dev.kord.rest.builder.message.EmbedBuilder
+import dev.kord.rest.builder.message.create.embed
+import dev.kord.rest.builder.message.modify.embed
+import org.koin.dsl.bind
+import org.quiltmc.community.cozy.modules.ama.data.AmaData
+import org.quiltmc.community.cozy.modules.ama.enums.QuestionStatusFlag
+
+public fun ExtensibleBotBuilder.ExtensionsBuilder.extAma(data: AmaData) {
+	loadModule { single { data } bind AmaData::class }
+
+	add(::AmaExtension)
+}
+
+public fun EmbedBuilder.questionEmbed(
+	interactionUser: User,
+	question: String?,
+	flag: QuestionStatusFlag,
+	flaggedBy: User? = null,
+	claimedOrSkippedBy: User? = null,
+	viaStage: Boolean? = null
+) {
+	author {
+		name = "${interactionUser.tag} (${interactionUser.id})"
+		icon = interactionUser.avatar?.cdnUrl?.toUrl()
+	}
+	description = question
+	color = when (flag) {
+		QuestionStatusFlag.ACCEPTED, QuestionStatusFlag.CLAIMED -> DISCORD_GREEN
+		QuestionStatusFlag.DENIED, QuestionStatusFlag.SKIPPED -> DISCORD_RED
+		QuestionStatusFlag.FLAGGED -> DISCORD_YELLOW
+		QuestionStatusFlag.ANSWERED -> DISCORD_BLURPLE
+		QuestionStatusFlag.NO_FLAG -> null
+	}
+
+	when (flag) {
+		QuestionStatusFlag.FLAGGED ->
+			if (flaggedBy != null) {
+				footer {
+					text = "Flagged by ${flaggedBy.tag}"
+					icon = flaggedBy.avatar?.cdnUrl?.toUrl()
+				}
+			}
+
+		QuestionStatusFlag.CLAIMED ->
+			if (claimedOrSkippedBy != null) {
+				footer {
+					text = "Claimed by ${claimedOrSkippedBy.tag}"
+					icon = claimedOrSkippedBy.avatar?.cdnUrl?.toUrl()
+				}
+			}
+
+		QuestionStatusFlag.SKIPPED ->
+			if (claimedOrSkippedBy != null) {
+				footer {
+					text = "Skipped by ${claimedOrSkippedBy.tag}"
+					icon = claimedOrSkippedBy.avatar?.cdnUrl?.toUrl()
+				}
+			}
+
+		QuestionStatusFlag.ANSWERED ->
+			if (viaStage == true && claimedOrSkippedBy != null) {
+				footer {
+					text = "Question answered via stage by ${claimedOrSkippedBy.tag}"
+				}
+			} else if (viaStage == false && claimedOrSkippedBy != null) {
+				footer {
+					text = "Question answered via text by ${claimedOrSkippedBy.tag}"
+				}
+			}
```

Comment:
Here it seems reasonable to extract the claim check out before the method. Maybe something like this?
```kt
if (claimedOrSkippedBy != null) {
    val answerMethod = if (viaStage) "stage" else "text"
    footer {
        text = "Question answered via $answerMethod by ${claimedOrSkippedBy.tag}"
    }
}
```